pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultiStageQueryThrottler.java

```diff
@@ -193,4 +204,24 @@ public int currentQueryServerThreads() {
   int availablePermits() {
     return _semaphore.availablePermits();
   }
+
+  @VisibleForTesting
+  int calculateMaxServerQueryThreads(PinotConfiguration config) {
+    Map<String, String> clusterConfigMap = _helixAdmin.getConfig(
+        _helixConfigScope,
+        Collections.singletonList(CommonConstants.Helix.CONFIG_OF_MULTI_STAGE_ENGINE_MAX_SERVER_QUERY_THREADS));
+    int maxServerQueryThreadsFromClusterConfig = Integer.parseInt(
+        clusterConfigMap.getOrDefault(
+            CommonConstants.Helix.CONFIG_OF_MULTI_STAGE_ENGINE_MAX_SERVER_QUERY_THREADS,
+            CommonConstants.Helix.DEFAULT_MULTI_STAGE_ENGINE_MAX_SERVER_QUERY_THREADS));
+    int maxServerQueryThreadsFromBrokerConfig = config.getProperty(
+            CommonConstants.Broker.CONFIG_OF_MSE_MAX_SERVER_QUERY_THREADS,
+            CommonConstants.Broker.DEFAULT_MSE_MAX_SERVER_QUERY_THREADS);
+
+    if (maxServerQueryThreadsFromClusterConfig <= 0 && maxServerQueryThreadsFromBrokerConfig <= 0) {
+      return 0;
+    }
+    return Math.min(maxServerQueryThreadsFromClusterConfig,
```

Comment:
If only one of them are positive, we want to pick the one positive. Currently it will end up picking the one negative